File: Java/ArrayString/41_maxConsecutiveOnes2.java
public int findMaxConsecutiveOnes(int[] nums) {
        int si =0, ei = 0, n = nums.length,len = -1, count =0 ;

        while(ei < n){
            if(nums[ei++] == 0) count++;
            while(count > 1){
                if(nums[si++] == 0) count--;
            }
            len = Math.max(len,ei - si);
        }
        return len;
    }

File: C++/ArrayString/41_maxConsecutiveOnes2.cpp
#include <bits/stdc++.h>
using namespace std;

int findMaxConsecutiveOnes(vector<int> &nums)
{
    // write your code here
    int maxLen = INT_MIN;
    int Zcount = 0, j = -1;
    for (int i = 0; i < nums.size(); i++)
    {
        if (nums[i] == 0)
            Zcount++;
        while (Zcount > 1)
        {
            j++;
            if (nums[j] == 0)
                Zcount--;
        }
        int len = i - j;
        if (len > maxLen)
            maxLen = len;
    }
    return maxLen;
}